core/src/main/resources/hudson/security/SecurityRealm/loginLink.jelly

```diff
@@ -26,7 +26,7 @@ THE SOFTWARE.
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler">
   <j:invokeStatic var="from" className="java.net.URLEncoder" method="encode">
-    <j:arg value="${if (request.session.attribute('from')!=null) request.session.getAttribute('from');  else if (request.getParameter('from')!=null) request.getParameter('from'); else if (request.requestURI=='/loginError' || request.requestURI=='/login') '/'; else request.requestURI;}"/>
+    <j:arg value="${if (request.getSession(false)!=null && request.session.attribute('from')!=null) request.session.getAttribute('from'); else if (request.getParameter('from')!=null) request.getParameter('from'); else if (request.requestURI==null || request.requestURI=='/loginError' || request.requestURI=='/login') '/'; else request.requestURI;}"/>
```

Comment:
Can we move the expression into a method of `SecurityRealm`? It will be much easier to understand and hopefully unittest. Or perhaps the whole `j:invokeStatic` thing.